packages/regenesis-surgery/scripts/utils.ts

```diff
@@ -17,3 +21,57 @@ export const hexStringEqual = (a: string, b: string): boolean => {
 
   return a.toLowerCase() === b.toLowerCase()
 }
+
+export const getSolc = (version: string, ovm?: boolean) => {
+  if (ovm) {
+    return solc.setupMethods(require(path.join(LOCAL_SOLC_DIR, version)))
+  }
+  return solc.setupMethods(
+    require(path.join(LOCAL_SOLC_DIR, `solc-emscripten-wasm32-${version}.js`))
+  )
+}
+
+export const solcInput = (contract: EtherscanContract) => {
+  // Create a base solc input object
+  const input = {
+    language: 'Solidity',
+    sources: {
+      file: {
+        // TODO: does this need the brackets?
+        content: contract.sourceCode,
+      },
+    },
+    settings: {
+      outputSelection: {
+        '*': {
+          '*': ['*'],
```

Comment:
Okay this includes `metadata` too I suppose. Now only thing is to preserve the `artifacts/build-info` for contracts to be uploaded to IPFS same as the Optimism contracts themselves see https://github.com/ethereum-optimism/optimism/pull/1534